src/crypto.rs

```diff
@@ -381,20 +369,152 @@ impl PrivateKey {
         let private_key_bytes = &key[..ED25519_PRIVATE_KEY_LENGTH];
         let public_key_bytes = &key[ED25519_PUBLIC_KEY_LENGTH..];
 
-        let key = Ed25519KeyPair::from_seed_and_public_key(private_key_bytes, public_key_bytes)
+        let private = Ed25519KeyPair::from_seed_and_public_key(private_key_bytes, public_key_bytes)
             .map_err(|err| Error::Encoding(err.to_string()))?;
 
         let public = PublicKey::new(
             KeyType::Ed25519,
             SignatureScheme::Ed25519,
             keyid_hash_algorithms,
-            key.public_key().as_ref().to_vec(),
+            private.public_key().as_ref().to_vec(),
+        )?;
+
+        Ok(Ed25519PrivateKey { private, public })
+    }
+}
+
+impl PrivateKey for Ed25519PrivateKey {
+    fn sign(&self, msg: &[u8]) -> Result<Signature> {
+        match &self.public.scheme {
```

Comment:
I don't think it's necessary to do this check. Instead, we'd only need to check this when we're constructing the `Ed25519PrivateKey` while passing in a public key, but I don't think we should have any codepaths that do that.

Reply:
Moved into a debug_assert